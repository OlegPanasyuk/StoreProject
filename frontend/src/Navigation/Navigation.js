import React, { Component } from 'react';
import { Nav, Navbar, Container } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
// import './Navigation.css';

export class Navigation extends Component {
    render() {
        let toggler = (
            <i className="fas fa-bars" style={{
                color: 'var(--light)'
            }}></i>
        );

        return (
            <Navbar expand="md" style={{
                backgroundColor: 'var(--primary)'
            }}>
                <Container>
                    <Navbar.Brand style={{
                        color: 'var(--light)'
                        
                    }}>
                        StoreProject
                    </Navbar.Brand>
                    <Navbar.Toggle style={{
                        border: 'none',
                        opacity: '0.8'
                    }}>
                        {toggler}
                    </Navbar.Toggle>
                    <Navbar.Collapse >
                        <Nav className=' mr-auto'>
                            <Nav.Item>
                                <NavLink to="/" className='p-3'>Home</NavLink>
                            </Nav.Item>
                            <Nav.Item>
                                <NavLink to={`/about`} className='p-3'>About</NavLink>
                            </Nav.Item>
                            <Nav.Item>
                                <NavLink to='/catalogue' className='p-3'>Catalogue</NavLink>
                            </Nav.Item>
                            <Nav.Item>
                                <NavLink to='/contacts' className='p-3'>Contacts</NavLink>
                            </Nav.Item>
                        </Nav>
                        {this.props.children}
                    </Navbar.Collapse>
                </Container>
            </Navbar>
        );
    }
}

Navigation.propTypes = {
    children: PropTypes.array
};

export default Navigation;
//