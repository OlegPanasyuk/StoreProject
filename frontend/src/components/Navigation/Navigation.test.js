import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Navigation from './Navigation';

configure({ adapter: new Adapter() });

function setupComponent(props = {

}) {
    const component = shallow(<Navigation {...props} />);
    return { component, props };
}

describe('Navigation ', () => {
    it('+++ component must be created', () => {
        const { component } = setupComponent();
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
