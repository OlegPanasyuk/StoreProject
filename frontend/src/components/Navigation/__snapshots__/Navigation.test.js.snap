// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation  +++ component must be created 1`] = `
<Bootstrap(undefined)
  expand="md"
  style={
    Object {
      "backgroundColor": "var(--primary)",
    }
  }
>
  <Container
    as="div"
    fluid={false}
  >
    <NavbarBrand
      style={
        Object {
          "color": "var(--light)",
        }
      }
    >
      StoreProject
    </NavbarBrand>
    <NavbarToggle
      as="button"
      label="Toggle navigation"
      style={
        Object {
          "border": "none",
          "opacity": "0.8",
        }
      }
    >
      <i
        className="fas fa-bars"
        style={
          Object {
            "color": "var(--light)",
          }
        }
      />
    </NavbarToggle>
    <NavbarCollapse>
      <Nav
        as="div"
        className=" mr-auto"
        fill={false}
        justify={false}
      >
        <NavItem
          as="div"
        >
          <NavLink
            className="p-3"
            to="/"
          >
            Home
          </NavLink>
        </NavItem>
        <NavItem
          as="div"
        >
          <NavLink
            className="p-3"
            to="/about"
          >
            About
          </NavLink>
        </NavItem>
        <NavItem
          as="div"
        >
          <NavLink
            className="p-3"
            to="/catalogue"
          >
            Catalogue
          </NavLink>
        </NavItem>
        <NavItem
          as="div"
        >
          <NavLink
            className="p-3"
            to="/contacts"
          >
            Contacts
          </NavLink>
        </NavItem>
      </Nav>
    </NavbarCollapse>
  </Container>
</Bootstrap(undefined)>
`;
